lib/core/enums/full_name_separator.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:thunder/thunder/bloc/thunder_bloc.dart';
+
+enum FullNameSeparator {
+  dot('name · instance.tld'),
+  at('name@instance.tld');
+
+  final String label;
+
+  const FullNameSeparator(this.label);
+}
+
+String generateUserFullName(BuildContext context, name, instance) {
+  final ThunderState thunderState = context.read<ThunderBloc>().state;
+  if (thunderState.userSeparator == FullNameSeparator.dot) {

```

Comment:
Just a stylistic syntactic thing, but I feel like using a `switch` here makes things a bit cleaner. Thoughts?

```dart
switch (thunderState.userSeparator) {
  case FullNameSeparator.dot:
    return '$name · $instance';
  case FullNameSeparator.at:
    return '$name@$instance';
  default:
    return '$name@$instance';
}
```

Reply:
Good idea! In fact, I think I'll take it a step further and use a switch expression instead of statement.

Reply:
> I think I'll take it a step further and use a switch expression instead of statement.

Even better, I didn't even know that was a thing 😅